### DISCUSSION TITLE
Testing model trained on fastMRI dataset on other datasets

### DISCUSSION THREAD

**Discussion by arp95** [facebookresearch/fastMRI#288] [Q&A]
Hi,

I was currently working on training deep learning models mapping low resolution to high resolution images. First of all, thanks for this great dataset, it is an amazing resource.
I trained the dataset on fastMRI train and val datasets. I used the tutorial of fastMRI to get the images from raw kspace to images where the notebook applied a series of operations inverse fourier transform, complex_abs to get the image. I trained the images using the fastMRI as i mentioned and normlized using the mean and std from the train dataset.

The question becomes I want to validate this model on images outside of fastMRI. The first issue is that once fastMRI applies complex_abs we get image pixels in floating point values ranging between -1 to 1 (can be more than 1 or -1 but the intensities are not in 100's as we usually expect image intensity to be). When i am using other mri datasets for validation they have pixel range between 0 and 600. how do i convert them to fastMRI data distribution range so that my trained model can be tested? I cant use my mean and std on train dataset as that was 0.000326 and 0.000557 respectively due to the range of values I got from fastMRI train dataset. Am i doing something wrong?

**Comment by mmuckley** [comment]
Hello @arp95, there are quite a few issues with extending to other MRI datasets. The data could have different preprocessing that gives different normalization values. The data could be from different scanners. The data could be with different pulse sequences. All of these could cause generalization issues.

It's not really possible to specify a procedure that will work in general. The best thing would be to work with an MRI physicist that understands how the systems work, and maybe be at an institution where you could acquire a small amount of data for validation and/or fine-tuning. Another option might be to ask the other people that released the datasets you're interested in how Siemens 2D imaging is related to their data. Lastly, if you post the datasets you're interested in here, maybe someone could give some advice that has experience with it.

**Comment by arp95** [comment]
So the fastMRI scans were acquired using Siemens scanners and if any dataset i am planning to validate this on is from Siemens should have same values?
For example I want to validate my fastMRI trained model on this dataset, https://www.kaggle.com/datasets/awsaf49/brats20-dataset-training-validation which is BRATS 2020 dataset on brain tumors. How should i go about it? Please advise @mmuckley

**Reply by mmuckley** [reply]
Hello @arp95, I don't know this specific dataset well enough to provide advice on it. If you're collaborating with an MR physicist, they could advise you.

**Comment by arp95** [comment]
Just one last question to clear my understanding over here @mmuckley 
So what i understand is for fastMRI we were given raw kspaces and fastMRI had their own inverse fourier transform and complex_abs function that gave me the images. Those images didnt have pixel values as other nifti images on the net do, they are mostly floating point between -1 to 1 i guess?
For other datasets to be compared with fastMRI trained models, i should have their raw kspaces so that I can apply fastMRI inverse fourier transform and complex_abs functions? Is that the right way? If I am given nifti images you are saying to collaborate with MR physicist right?

**Reply by mmuckley** [reply]
Hello @arp95, it is not really about the mathematical operations, but about the physical processes that give rise to the raw data. Depending on these physical processes (pulse sequence, scanner, sampling), there may be a wide array of corrections to apply. MR physicists spend years studying all of these cases and would be able to provide the best advice.

One thing I'm noticing is that you mentioned nifti files. Are these already-reconstructed images? The fastMRI dataset is primarily meant for the task of reconstruction, not tumor segmentation. Please be aware of any [data crimes](https://www.pnas.org/doi/abs/10.1073/pnas.2117203119) that may be associated with any approach that you are researching.

Shimron, Efrat, et al. "Implicit data crimes: Machine learning bias arising from misuse of public data." Proceedings of the National Academy of Sciences 119.13 (2022): e2117203119.

**Reply by mmuckley** [reply]
@arp95, sorry, I can only provide assistance on the fastMRI data or the fastMRI repository.

**Reply by arp95** [reply]
@mmuckley For fastMRI dataset do we need to do bias field correction to ensure when we perform root mean square from multi-coil no issue takes place?

**Reply by mmuckley** [reply]
@arp95 for calculating metrics we don't use any bias field.

If you are displaying images to radiologists at your institution, you should apply whatever post-processing they request. Our radiologists generally didn't request bias field correction.

**Reply by arp95** [reply]
@mmuckley so you mean the way the multicoil data was collected from fastMRI did not need this biad field correction as usually i have seen MRI images having this preprocessing step.

**Reply by mmuckley** [reply]
@arp95 it depends on what you mean by "bias field". There is some B0/B1 inhomogeneity that can lead to certain parts of the image being brighter or darker, which bothers some radiologists. This is usually corrected via **post-processing**.

To my knowledge there is no pre-processing for bias field correction.

Please note that the task for fastMRI data is **reconstruction**, not analysis. From the perspective of **reconstruction**, bias-field correction is a **post-processing** step. From the perspective of analysis of already-reconstructed images, bias-field correction is a pre-processing step.

**Reply by arp95** [reply]
got it. thanks!
